pallets/compliance-manager/src/lib.rs

```diff
@@ -780,33 +779,41 @@ impl<T: Trait> ComplianceManagerTrait<T::Balance> for Module<T> {
         from_did_opt: Option<IdentityId>,
         to_did_opt: Option<IdentityId>,
         _value: T::Balance,
+        treasury_did: Option<IdentityId>,
     ) -> StdResult<u8, &'static str> {
-        // Transfer is valid if ALL receiver AND sender rules of ANY asset rule are valid.
+        let (treasury_sender, treasury_receiver) = if treasury_did.is_some() {
+            (treasury_did == from_did_opt, treasury_did == to_did_opt)
+        } else {
+            (false, false)
+        };
+        // Transfer is valid if ALL receiver AND sender rules of ANY asset rule are valid. An
+        // exception is made for the cases when either the sender or the receiver is the treasury
+        // DID, which covers issuance and redemption transactions.
         let asset_rules = Self::asset_rules(ticker);
-        if asset_rules.is_paused {
+        if asset_rules.is_paused
+            || (asset_rules.rules.len() == 0 && (from_did_opt == None && treasury_receiver)
+                || (treasury_sender && to_did_opt == None))
+        {
```

Comment:
Happy to approve, but you could get rid of these special cases by doing:
```
        if asset_rules.is_paused
        {
            return Ok(ERC1400_TRANSFER_SUCCESS);
        }
        let mut rule_satisfied = true;
        for active_rule in asset_rules.rules {
            if let Some(from_did) = from_did_opt {
                rule_satisfied = treasury_sender
                    || Self::are_all_rules_satisfied(ticker, from_did, &active_rule.sender_rules);
                if !rule_satisfied {
                    // Skips checking receiver rules because sender rules are not satisfied.
                    continue;
                }
            }
            if let Some(to_did) = to_did_opt {
                rule_satisfied = treasury_receiver
                    || Self::are_all_rules_satisfied(ticker, to_did, &active_rule.receiver_rules);
            }
        }
        if rule_satisfied {
            return Ok(ERC1400_TRANSFER_SUCCESS);
        }
```